.circleci/local_publish_helpers.sh

```diff
@@ -102,33 +102,35 @@ function verifyPkgCli {
     echo "Sizes in bytes"
     wc -c out/*
 
-    linux_compressed_binary_threshold_in_bytes=$((200 * 1024 * 1024))
-    linux_compressed_binary_size=$(wc -c out/amplify-pkg-linux-x64.tgz | awk '{print $1}')
-    if (( linux_compressed_binary_size > linux_compressed_binary_threshold_in_bytes )); then
-      echo "Linux compressed binary size has grown over $linux_compressed_binary_threshold_in_bytes bytes"
-      exit 1
-    fi
-
-    macos_compressed_binary_threshold_in_bytes=$((200 * 1024 * 1024))
-    macos_compressed_binary_size=$(wc -c out/amplify-pkg-macos-x64.tgz | awk '{print $1}')
-    if (( macos_compressed_binary_size > macos_compressed_binary_threshold_in_bytes )); then
-      echo "MacOS compressed binary size has grown over $macos_compressed_binary_threshold_in_bytes bytes"
-      exit 1
-    fi
+    function verifySinglePkg {
+      binary_name=$1
+      compressed_binary_name=$2
+      binary_threshold_in_bytes=$3
+
+      # Compressed binary size is not deterministic enough to have stricter threshold.
+      # I.e. it depends on how compression algorithm can compress bytecode and there are cases where compressed size
+      # grows even if uncompressed size drops. We don't have control on bytecode and compression.
+      # Therefore we check if compression gets past half of original size as sanity check.
+      compressed_binary_threshold_in_bytes=$((binary_threshold_in_bytes/2))
+
+      binary_size=$(wc -c out/$binary_name | awk '{print $1}')
+      compressed_binary_size=$(wc -c out/$compressed_binary_name | awk '{print $1}')
+
+      if (( binary_size > binary_threshold_in_bytes )); then
```

Comment:
nit, does this do string comparison or is it typed?

Reply:
Arithmetic comparison: http://mywiki.wooledge.org/ArithmeticExpression

Reply:
`if (( )) ` is [arithmetic context](http://mywiki.wooledge.org/ArithmeticExpression) , as opposed to `if [ ]` or `if [[ ]]`.

So. This compares numbers.